apps/balancer-tools/src/app/apr/(utils)/calculateRoundAPR.ts

```diff
@@ -16,27 +16,53 @@ export async function calculatePoolStats({
   roundId: string;
   poolId: string;
 }) {
-  // TODO:  BAL-646 aggregate historical pool APR when roundId is not provided
+  // TODO: BAL-646 aggregate historical pool APR when roundId is not provided
   const round = Round.getRoundByNumber(roundId);
   const pool = new Pool(poolId);
 
   const endRoundBlockNumber = await getBlockNumberByTimestamp(
-    pool.gauge?.network,
-    round.endDate,
+    pool.gauge?.network ?? 1,
+    round.activeRound ? round.startDate : round.endDate,
   );
 
-  const [balPriceUSD, tvl, votingShare] = await Promise.all([
-    getBALPriceByRound(round),
-    // TODO: BAL-648 TVL from the selected round
-    BalancerAPI.getPoolTotalLiquidityUSD(
-      pool.gauge?.network || 1,
-      pool.id,
-      endRoundBlockNumber,
-    ),
-    getPoolRelativeWeight(poolId, round.endDate.getTime() / 1000),
-  ]);
-
-  const apr = calculateRoundAPR(round, votingShare, tvl, balPriceUSD) * 100;
+  let balPriceUSD, tvl, votingShare;
+
+  try {
+    balPriceUSD = await getBALPriceByRound(round);
+  } catch (error) {
+    balPriceUSD = 0;
+  }
+
+  try {
+    tvl = await pools
+      .gql(String(pool.gauge?.network) ?? 1)
+      .PoolWhereBlockNumber({
+        blockNumber: endRoundBlockNumber,
+        poolId,
+      })
+      .then((res) => {
+        if (!res.pool?.totalLiquidity) {
+          throw new Error("Failed to fetch totalLiquidity.");
+        }
+        return res.pool?.totalLiquidity;
+      });
+  } catch (error) {
+    tvl = 0;
+  }
```

Comment:
if you're throwing here to catch below why don't you just set tvl = 0 after getting the response?

Reply:
We have a task for that https://linear.app/bleu-llc/issue/BAL-655/handle-0-values-on-pool-stats